app/models/submission_match.rb

```diff
@@ -25,7 +25,16 @@ def term
   end
 
   def query
-    { state: params[:state].presence, user_id: params[:user].presence }.compact
+    attributes = {
+      state: params[:state].presence,
+      user_id: params[:user].presence
+    }.compact
+    attributes.merge!(assigned_to: assigned_to) if filtering_by_assigned_to?
```

Comment:
**question [I'm learning]:** 

Is there a construct in ruby that would allow us to inline the toggling of the `assigned_to` attribute in the `attributes` assignment? I know this doesn't work, but something like....

```
    attributes = {
      state: params[:state].presence,
      user_id: params[:user].presence,
      assigned_to: assigned_to if filtering_by_assigned_to?
    }.compact
```

_Comment format based on [conventionalcomments.org](https://conventionalcomments.org/)_

Reply:
Here's the rub: we want to conditionally suppress a key from this hash. I'm not aware of another way other than what you see here. Let's revisit our possible choices in the UI:

* all => suppress the assigned_to key from the hash altogether
* none => add the assigned_to key with a value of nil so you can find the ones that are unassigned
* alice (or whatever) => return only the submissions with a matching assignee

So it's that first that's causing this slightly less cool approach here. 🤷